doc/classes/DirAccess.xml

```diff
@@ -204,6 +204,14 @@
 				Returns the available space on the current directory's disk, in bytes. Returns [code]0[/code] if the platform-specific method to query the available space fails.
 			</description>
 		</method>
+		<method name="is_case_sensitive" qualifiers="const">
+			<return type="bool" />
+			<param index="0" name="path" type="String" />
+			<description>
+				Returns [code]true[/code] if file system or directory use case sensitive file names.
+				[b]Note:[/b] This method is implemented on macOS and Windows.
```

Comment:
```suggestion
				[b]Note:[/b] This method is implemented on macOS and Windows. On other platforms, it always returns [code]true[/code], as they are case sensitive.
```

Reply:
I would skip this note entirely.

Case sensitivity is filesystem related rather than OS related.

Reply:
> Case sensitivity is filesystem related rather than OS related.

The checks are OS specific and only implemented for macOS and Windows (also, in case of Windows it's directory specific).

Reply:
Oh, maybe I'm lost in translation. Does "as they are case sensitive" mean "it returns true because they are case sensitive" or "it returns true and users should treat it as case sensitive"? The latter makes sense. I thought it was the former, so I think it conflicts with the first sentence.

Reply:
Yep, I'm not entirely sure about "as they are case sensitive" part either.

Reply:
My assumption was that filesystems used in Linux, Android, iOS and Web are case sensitive. But if they're not all, then yeah it's incorrect.

Reply:
> But if they're not all, then yeah it's incorrect.

Usually they are, but on Linux it should be possible to have a non-case-sensitive file system as well (I have no idea how to do the check). So I guess it's probably better to remove this part.

Reply:
> Usually they are, but on Linux it should be possible to have a non-case-sensitive file system as well (I have no idea how to do the check).

Yeah, if you mount a case-insensitive file system on Linux (ex FAT or NTFS) it will be case-insensitive under that directory. I also have no idea how to check :-)

Reply:
Hm it might depend on the drivers / configuration, I don't see it case insensitive on NTFS from Linux here:
```
$ cd /media/windows
$ touch notice.txt
$ ls -l [Nn]*
-rwxrwxrwx 1 root root    0 Oct  9 14:34 notice.txt*
-rwxrwxrwx 1 root root 5472 Jun  9  2021 NOTICE.txt*
```

But indeed, even ext4 can be configured to be case insensitive it seems: https://www.collabora.com/news-and-blog/blog/2020/08/27/using-the-linux-kernel-case-insensitive-feature-in-ext4/

Reply:
> I have no idea how to do the check

For the reference, I have not found a universal way to do it on Linux, EXT4 specific check seems to be possible (have not checked if it works), but I'm not sure if it's worth adding:
```c++
        long flags;
        if (ioctl(fd, EXT2_IOC_GETFLAGS, &flags) < 0) {
            close(fd);
            return true;
        }
        return !(flags & EXT4_CASEFOLD_FL);
```

Reply:
It could make sense to add if it's relatively simple, and we can document it's only implemented for EXT4 (which is still the most common). I can test it. Someone might come up with a btrfs implementation if it matters.

Reply:
Tested it with EXT4 and is works fine (same code should also support F2FS partitions) - https://github.com/godotengine/godot/pull/83266

Other file systems don't have a similar flag, so if case folding is supported it's probably implemented differently.